e2e/rapid_test.go

```diff
@@ -0,0 +1,598 @@
+package e2e
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/0xPolygon/pbft-consensus"
+	"go.opentelemetry.io/otel/trace"
+	"golang.org/x/sync/errgroup"
+	"io"
+	"log"
+	"pgregory.net/rapid"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+)
+
+const waitDuration = 50 * time.Millisecond
+
+type fakeTransport struct {
+	nodes      []*pbft.Pbft
+	GossipFunc func(ft *fakeTransport, msg *pbft.MessageReq) error
+}
+
+func (ft *fakeTransport) Gossip(msg *pbft.MessageReq) error {
+	if ft.GossipFunc != nil {
+		return ft.GossipFunc(ft, msg)
+	}
+
+	for _, node := range ft.nodes {
+		if msg.From != node.GetValidatorId() {
+			node.PushMessage(msg.Copy())
+		}
+	}
+	return nil
+}
+
+type BackendFake struct {
```

Comment:
Could we reuse some of the e2e backend logic here as well (embedding it and overriding the functions which aren't generic enough for property based testing framework)?

Reply:
I suggest to resolve it as a part of https://linear.app/matic/issue/POS3-184/property-tests-toolkit-refactoring